SoftLayer/managers/dedicated_host.py

```diff
@@ -0,0 +1,333 @@
+        mask = '''
+        items[
+            id,
+            description,
+            prices,
+            itemCategory[categoryCode]
+        ],
+        regions[location[location[priceGroups]]]
+        '''
+
+        package_keyname = 'DEDICATED_HOST'
+
+        package = self.ordering_manager.get_package_by_key(package_keyname,
+                                                           mask=mask)
+
+        if package is None:
+            raise SoftLayer.SoftLayerError("Ordering package not found")
+
+        return package
+
+    def _get_location(self, regions, datacenter):
+        """Get the longer key with a short location(datacenter) name."""
+        for region in regions:
+            # list of locations
+            if region['location']['location']['name'] == datacenter:
+                return region
+
+        raise SoftLayer.SoftLayerError("Could not find valid location for: '%s'"
+                                       % datacenter)
+
+    def get_create_options(self):
+        """Returns valid options for ordering a dedicated host."""
+
+        package = self._get_package()
+        # Locations
+        locations = []
+        for region in package['regions']:
+            locations.append({
+                'name': region['location']['location']['longName'],
+                'key': region['location']['location']['name'],
+            })
+        dedicated_host = []
+        for item in package['items']:
+            if item['itemCategory']['categoryCode'] == \
+                    'dedicated_virtual_hosts':
+                dedicated_host.append({
+                    'name': item['description'],
+                    'key': item['id'],
+                })
+
+        return {
+            'locations': locations,
+            'dedicated_host': dedicated_host,
+        }
+
+    def _get_price(self, package):
+        """Returns valid price for ordering a dedicated host."""
+
+        for price in package['prices']:
+            if price.get('locationGroupId') is '':
+                return price['id']
+
+        raise SoftLayer.SoftLayerError(
+            "Could not find valid price")
+
+    def _get_item(self, package):
+        """Returns the item for ordering a dedicated host."""
+        description = '56 Cores X 242 RAM X 1.2 TB'
+
+        for item in package['items']:
+            if item['description'] == description:
+                return item
+
+        raise SoftLayer.SoftLayerError("Could not find valid item for: '%s'"
+                                       % description)
+
+    def _get_backend_router(self, locations):
+        """Returns valid router options for ordering a dedicated host."""
+        mask = '''
+            id,
+            hostname
+        '''
+
+        if locations is not None:
+            for location in locations:
+                if location['locationId'] is not None:
+                    loc_id = location['locationId']
+                    host = {
```

Comment:
I can see this being problematic if more dedicated host sizes are added. This should be calculated from the package or supplied by the CLI I think. 

Unless its possible to not supply this data at all and just get a list of routers without specifying a template